tests/unit/commands/local/lib/test_sam_api_provider.py

```diff
@@ -1872,6 +1872,111 @@ def test_extract_serverless_api_extracts_default_authorizer(
 
         api_collector_mock.set_default_authorizer.assert_called_with(logical_id_mock, authorizer_name)
 
+    @parameterized.expand(
+        [("when enabled ignores ignores authorizers", True), ("when disabled extracts authorizers as normal", False)]
+    )
+    @patch("samcli.lib.providers.sam_api_provider.SamApiProvider._extract_authorizers_from_props")
+    @patch("samcli.lib.providers.sam_api_provider.CfnBaseApiProvider.extract_swagger_route")
+    @patch("samcli.lib.providers.sam_api_provider.LOG.debug")
+    def test_extract_from_serverless_http_with_disable_authorizer_flag(
+        self,
+        _,
+        disable_authorizer,
+        mock_debug_log: Mock,
+        mock_extract_swagger_route: Mock,
+        mock_extract_authorizers_from_props: Mock,
+    ):
+        provider = SamApiProvider()
+        template = {
+            "Properties": {
+                "DefinitionBody": "mock-body",
+                "DefinitionUri": "mock-uri",
+                "StageName": "stage-name",
+                "StageVariables": {},
+                "Auth": {"DefaultAuthorizer": "Authorizer"},
+            }
+        }
+        mock_collector = Mock()
+        provider._extract_from_serverless_http(
+            stack_path="/",
+            logical_id="id",
+            api_resource=template,
+            collector=mock_collector,
+            disable_authorizer=disable_authorizer,
+        )
+
+        if disable_authorizer:
+            mock_debug_log.assert_called_with("authorizer not found or disabled, returning early")
+            mock_collector.set_default_authorizer.assert_not_called()
+            mock_extract_authorizers_from_props.assert_not_called()
+
+        if not disable_authorizer:
+            mock_debug_log.assert_not_called()
```

Comment:
Thoughts about asserting log output? I think the previous comment I had noted that we could test the logic methods being called instead of debug logs that way we are testing the implementation.